pkg/controller/acmeorders/sync.go

```diff
@@ -612,13 +621,26 @@ func (c *controller) fetchCertificateData(ctx context.Context, cl acmecl.Interfa
 		return nil
 	}
 
-	// If the Order state has actually changed and we've not observed it,
-	// update the order status and let the change in the resource trigger
-	// a resync
+	return c.syncCertificateDataWithOrder(ctx, cl, *acmeOrder, o, issuer)
+}
+
+func (c *controller) syncCertificateDataWithOrder(ctx context.Context, cl acmecl.Interface, acmeOrder acmeapi.Order, o *cmacme.Order, issuer cmapi.GenericIssuer) error {
+	log := logf.FromContext(ctx)
+	// Certificate data can only be fetched for a valid order
 	if acmeOrder.Status != acmeapi.StatusValid {
 		return nil
 	}
 
+	if issuer.GetSpec().ACME != nil && issuer.GetSpec().ACME.PreferredChain != "" {
+		found, altCerts, err := getAltCertChain(ctx, cl, acmeOrder.CertURL, issuer.GetSpec().ACME.PreferredChain)
+		if err != nil {
+			return err
+		}
+		if found {
+			return c.storeCertificateOnStatus(ctx, o, altCerts)
+		}
+
+	}
```

Comment:
Would it be possible to move the change to `FetchCert` to another PR? 😇

Reply:
Is it making this PR hard to review? I'm just concious that it might get lost in the backlog and seem less comprehensible on its own as it _is_ related to a scenario where valid order without a cert gets reconciled.

Reply:
Never mind, let's keep it in the PR, it doesn't really obfuscate things.